autoload/codi.vim

```diff
@@ -388,12 +395,20 @@ function! s:codi_do_update()
   let bufnr = bufnr('%')
 
   " Build input
-  let input = join(getline('^', '$'), "\n")
+  if has('win32')
+    let input = join(getline('^', '$'), "\r\n")
+  else
+    let input = join(getline('^', '$'), "\n")
+  endif
   if has_key(i, 'rephrase')
     let input = i['rephrase'](input)
   endif
   if has_key(i, 'quitcmd')
-    let input = input."\n".i['quitcmd']."\n"
+    if has('win32')
+      let input = input."\r\n".i['quitcmd']."\r\n"
+    else
+      let input = input."\n".i['quitcmd']."\n"
+    endif
```

Comment:
Instead of writing two blocks with similar structure, how about defining a variable for the new line sequence?

```
  let newline = has('win32') ? "\r\n" : "\n"
  let input = join(getline('^', '$'), newline)
  if has_key(i, 'rephrase')
    let input = i['rephrase'](input)
  endif
  if has_key(i, 'quitcmd')
    let input = input.newline.i['quitcmd'].newline
```